pandas/io/parsers/readers.py

```diff
@@ -1515,6 +1515,12 @@ def _clean_options(
             if engine == "c" and sep == r"\s+":
                 result["delim_whitespace"] = True
                 del result["delimiter"]
+            elif engine == "pyarrow" and sep == r"\s+":
+                fallback_reason = (
+                    f"the '{engine}' engine does not support separators "
```

Comment:
Could you add a test for this?

Reply:
In which file and class would you like to add the test?

Reply:
Is class TestTextReader in file test_textreader.py acceptable?

Reply:
`pandas/tests/io/parser/common/test_common_basic.py`